kms/api/functions.py

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
+import argparse
+import base64
+
+# Imports the Google APIs client library
+from googleapiclient import discovery
+
+
+# [START kms_create_keyring]
+def create_keyring(project_id, location, keyring):
+
+    # Creates an API client for the KMS API.
+    kms_client = discovery.build('cloudkms', 'v1beta1')
+
+    # The resource name of the location associated with the KeyRing.
+    parent = 'projects/{}/locations/{}'.format(project_id, location)
+
+    # Create KeyRing
+    request = kms_client.projects().locations().keyRings().create(
+        parent=parent, body={}, keyRingId=keyring)
+    response = request.execute()
+
+    print 'Created KeyRing {}.'.format(response["name"])
+# [END kms_create_keyring]
+
+
+# [START kms_create_cryptokey]
+def create_cryptokey(project_id, location, keyring, cryptokey):
+
+    # Creates an API client for the KMS API.
+    kms_client = discovery.build('cloudkms', 'v1beta1')
+
+    # The resource name of the KeyRing associated with the CryptoKey.
+    parent = 'projects/{}/locations/{}/keyRings/{}'.format(
+        project_id, location, keyring)
+
+    # Create a CryptoKey for the given KeyRing.
+    request = kms_client.projects().locations().keyRings().cryptoKeys().create(
+        parent=parent, body={"purpose": 'ENCRYPT_DECRYPT'},
+        cryptoKeyId=cryptokey)
+    response = request.execute()
+
+    print 'Created CryptoKey {}.'.format(response["name"])
+# [END kms_create_cryptokey]
+
+
+# [START kms_encrypt]
+def encrypt(project_id, location, keyring, cryptokey, infile, outfile):
+
+    # Creates an API client for the KMS API.
+    kms_client = discovery.build('cloudkms', 'v1beta1')
+
+    # The resource name of the CryptoKey.
+    name = 'projects/{}/locations/{}/keyRings/{}/cryptoKeys/{}'.format(
+        project_id, location, keyring, cryptokey)
+
+    # Read text from input file.
+    i = open(infile, 'r')
```

Comment:
Use context managers when opening files, and use `io.open` instead of `open`.

```python
with io.open(plaintext_file_name, 'r') as plaintext_file:
    plaintext = plaintext_file.read()
```